pennylane/ops/op_math/exp.py

```diff
@@ -292,6 +319,9 @@ def eigvals(self):
         """
         base_eigvals = math.convert_like(self.base.eigvals(), self.coeff)
         base_eigvals = math.cast_like(base_eigvals, self.coeff)
+        if qml.math.size(self.scalar) > 1:
```

Comment:
Out of curiosity: Above you check for broadcasting with `ndim`, here with `size`. I suppose the check with `size` here would declare a scalar of shape `(1,)` to be non-broadcasted? That would break with current convention in PennyLane, as far as I can tell.

Reply:
You're right, we should use `ndim` everywhere 👍

Reply:
Changed!